canvasapi/canvas.py

```diff
@@ -1184,6 +1184,32 @@ def get_user_participants(self, appointment_group, **kwargs):
             _kwargs=combine_kwargs(**kwargs),
         )
 
+    def graphql(self, query, variables):
+        """
+        Makes a GraphQL formatted requeset to Canvas
+
+        :calls: `POST /api/graphql \
+        <https://canvas.instructure.com/doc/api/file.graphql.html>`_
+
+        :param query: The GraphQL query to execute
+        :type query: str
+
+        :param variables: Variable values as required by the supplied query
+        :type variables: dict
+
+        :rtype: dict
+        """
+        response = self.__requester.request(
+            "POST",
+            "graphql",
+            _apiv="",
+            headers={"Content-Type": "application/json"},
+            _kwargs=[("query", query), ("variables", variables)],
```

Comment:
Rather than constructing kwargs manually, I think using `canvasapi.util.combine_kwargs` here might make this more consistent with the rest of the library:

```python
    def graphql(self, **kwargs):
        """
        Makes a GraphQL formatted request to Canvas.

        :calls: `POST /api/graphql <https://canvas.instructure.com/doc/api/file.graphql.html>`_

        :rtype: dict
        """
        response = self.__requester.request(
            "POST",
            _url=self.__requester.base_url + "api/graphql",
            headers={"Content-Type": "application/json"},
            json=True,
            _kwargs=combine_kwargs(**kwargs),
        )

        return response.json()
```

This _should_ enable us to call this with kwargs directly:
```python
canvas.graphql(query=
   """
   query courseInfo($courseId: ID!) {
       course(id: $courseId) {
        id
        _id
        name
       }
     }
    """, variables={"courseId": 1})
```

I'm not sure what implications sending this as JSON instead of form data will have, but historically we've sent parameters like `variables[courseId]=1` using the combine_kwargs tuple recipe `variables={"courseId": 1}`. Something to think about.

Reply:
Ok I can try this. There was only two args query and variables that were supported by this endpoint.

Reply:
When I'm running this through combine_kwargs it's making the dict
` ('variables[termid]', 125)`
for this endpoint it seems to needs to stay as

`('variables', {'termid': 125})`

We'd need some way to stop combined_kwargs from doing anything with a dict. I can commit something maybe you have a better idea?

Reply:
If we can guarantee that `query` and `variables` are the only params, I don't mind constructing `_kwargs` manually. `combine_kwargs` is gonna screw with a lot of things people could potentially pass in.

This endpoint is fundamentally different from most of our endpoints, so I don't mind being a little inconsistent here.

Reply:
Yeah, it's either we
* Have fixed args here and construct it manually
* Use kwargs but pop them out and construct it manually
* Change combine_kwargs to avoid messing with this (I did a commit for this but am happy to switch it out as it seems hacky and only for this)

Reply:
`query` and `variables` as fixed args, with a catch-all `**kwargs` at the end makes the most sense to me.

Would want to run `combine_kwargs` against `kwargs` first, then add `query` and `variables` to it before passing to `_kwargs`.

I'm not a fan of 3. Like you said, it seems hacky.

Reply:
Okay, I made this change